media-gfx/openvdb/openvdb-11.0.0.ebuild

```diff
@@ -0,0 +1,247 @@
+)
+
+cuda_set_CUDAHOSTCXX() {
+	local compiler
+	tc-is-gcc && compiler="gcc"
+	tc-is-clang && compiler="clang"
+	[[ -z "$compiler" ]] && die "no compiler specified"
+
+	local package="sys-devel/${compiler}"
+	local version="${package}"
+	local CUDAHOSTCXX_test
+	while
+		CUDAHOSTCXX="${CUDAHOSTCXX_test}"
+		version=$(best_version "${version}")
+		if [[ -z "${version}" ]]; then
+			if [[ -z "${CUDAHOSTCXX}" ]]; then
+				die "could not find supported version of ${package}"
+			fi
+			break
+		fi
+		CUDAHOSTCXX_test="$(
+			dirname "$(
+				realpath "$(
+					which "${compiler}-$(echo "${version}" | grep -oP "(?<=${package}-)[0-9]*")"
+				)"
+			)"
+		)"
+		version="<${version}"
+	do ! echo "int main(){}" | nvcc "-ccbin ${CUDAHOSTCXX_test}" - -x cu &>/dev/null; done
+
+	export CUDAHOSTCXX
+}
+
+cuda_get_host_arch() {
+	[[ -z "${CUDAARCHS}" ]] && einfo "trying to determine host CUDAARCHS"
+	: "${CUDAARCHS:=$(__nvcc_device_query)}"
+	einfo "building for CUDAARCHS = ${CUDAARCHS}"
+
+	export CUDAARCHS
+}
+
+pkg_setup() {
+	use ax && llvm_pkg_setup
+	use python && python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	cmake_src_prepare
+
+	sed -e 's|/usr/local/bin/python|/usr/bin/python|' \
+		-i "${S}"/openvdb/openvdb/python/test/TestOpenVDB.py || die
+}
+
+src_configure() {
+	local myprefix="${EPREFIX}/usr/"
+
+	local version
+	if use abi8-compat; then
+		version=8
+	elif use abi9-compat; then
+		version=9
+	elif use abi10-compat; then
+		version=10
+	elif use abi11-compat; then
+		version=11
+	else
+		die "OpenVDB ABI version is not compatible"
+	fi
+
+	local mycmakeargs=(
+		-DCMAKE_INSTALL_DOCDIR="share/doc/${PF}/"
+		-DOPENVDB_ABI_VERSION_NUMBER="${version}"
+		-DOPENVDB_BUILD_DOCS=$(usex doc)
+		-DOPENVDB_BUILD_UNITTESTS=$(usex test)
+		-DOPENVDB_BUILD_VDB_LOD=$(usex utils)
+		-DOPENVDB_BUILD_VDB_RENDER=$(usex utils)
+		-DOPENVDB_BUILD_VDB_TOOL=$(usex utils)
+		-DOPENVDB_BUILD_VDB_VIEW=$(usex utils)
+		-DOPENVDB_CORE_SHARED=ON
+		-DOPENVDB_CORE_STATIC=$(usex static-libs)
+		-DOPENVDB_ENABLE_RPATH=OFF
+		-DUSE_AX=$(usex ax)
+		-DUSE_BLOSC=$(usex blosc)
+		-DUSE_CCACHE=OFF
+		-DUSE_COLORED_OUTPUT=ON
+		# OpenEXR is only needed by the vdb_render tool and defaults to OFF
+		-DUSE_EXR=$(usex utils)
+		-DUSE_IMATH_HALF=ON
+		-DUSE_LOG4CPLUS=ON
+		-DUSE_NANOVDB=$(usex nanovdb)
+		-DNANOVDB_USE_OPENVDB=$(usex nanovdb)
+		-DUSE_PKGCONFIG=ON
+		# PNG is only needed by the vdb_render tool and defaults to OFF
+		-DUSE_PNG=$(usex utils)
+		-DUSE_TBB=ON
+		-DUSE_ZLIB=$(usex ax ON $(usex blosc))
+	)
+
+	if use abi9-compat; then
+		mycmakeargs+=( -DOPENVDB_USE_DEPRECATED_ABI_9=ON )
+	elif use abi8-compat; then
+		mycmakeargs+=( -DOPENVDB_USE_DEPRECATED_ABI_8=ON )
+	fi
+
+	if use ax; then
+		mycmakeargs+=(
+			-DOPENVDB_AX_STATIC=$(usex static-libs)
+			-DOPENVDB_BUILD_AX_UNITTESTS=OFF # FIXME: log4cplus init and other errors
+			-DOPENVDB_BUILD_VDB_AX=$(usex utils)
+		)
+	fi
+
+	if use nanovdb; then
+		mycmakeargs+=(
+			# -DNANOVDB_BUILD_UNITTESTS=$(usex test)
```

Comment:
Why is the unit tests for nanovdb disabled?

Reply:
Iirc because the failed in src_test but not outside of it. Will recheck. The splitting of openvdb and nanovdb might be worth looking into. Only nanovdb has cuda support.

Reply:
Did you re-check this now or not?